src/task_dynamics/Reference.cpp

```diff
@@ -71,14 +71,16 @@ namespace tvm::task_dynamics
 
   void Reference::Impl::setReference(const FunctionPtr& ref)
   {
-    if (ref)
+    if (!ref)
+    {
+      throw std::runtime_error("[task_dynamics::Reference::Impl] You cannot pass a nullptr as reference function.");
+    }
+    else
```

Comment:
No need for `else` since the function throws if the pointer is null

Reply:
Done in b8863d5